src/backend/distributed/executor/multi_executor.c

```diff
@@ -617,12 +617,22 @@ IsLocalReferenceTableJoinPlan(PlannedStmt *plan)
 		return false;
 	}
 
-	foreach(oidCell, plan->relationOids)
+	foreach(rangeTableCell, plan->rtable)
 	{
-		Oid relationId = lfirst_oid(oidCell);
+		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
 		bool onlySearchPath = false;
 
-		if (RelationIsAKnownShard(relationId, onlySearchPath))
+		if (rangeTableEntry->rtekind == RTE_FUNCTION)
+		{
+			return false;
```

Comment:
why return false here? What if we a join between a local table, a reference table and a function?

Reply:
To match distributed_planner.c's `IsLocalReferenceTableJoin()`. Joins between local, reference, and functions will error out in planner. We can remove that restriction in both planner and executor, but maybe in another PR.

Removing the check here, although I think it won't change any behaviour.